bc_obps/reporting/service/report_product_service.py

```diff
@@ -64,3 +70,92 @@ def get_production_data(cls, report_version_id: int, facility_id: UUID) -> Query
+            report_version_id=report_version_id,
+            product_id=product_id,
+        )
+        product_annual_amount_sum = records.aggregate(production_sum=Sum('annual_production'))
+        apr_dec_sum = records.aggregate(production_sum=Sum('production_data_apr_dec'))
+
+        return {
+            "annual_amount": product_annual_amount_sum['production_sum'] or 0,
+            "apr_dec": apr_dec_sum['production_sum'] or 0,
+        }
+
+    @staticmethod
+    def get_reporting_only_allocated(report_version_id: int, product_id: int) -> Decimal | int:
+        # CO2 emissions from excluded woody biomass, Other emissions from excluded biomass, Emissions from excluded non-biomass, Fugitive emissions, Venting emissions — non-useful
+        reporting_only_category_ids = [10, 11, 12, 2, 7]
+        reporting_only_allocated = ReportProductService.get_allocated_emissions_by_report_product_emission_category(
+            report_version_id, product_id, reporting_only_category_ids
+        )
+        fog_records = ReportProductEmissionAllocation.objects.filter(
+            report_version_id=report_version_id,
+            report_product__product_id=39,  # Special Fat, Oil & Grease product
+        )
+        fog_allocated_amount = fog_records.aggregate(allocated_sum=Sum('allocated_quantity'))
+        return reporting_only_allocated + (fog_allocated_amount['allocated_sum'] or 0)
+
+    @classmethod
+    def get_compliance_production_data(cls, report_version_id: int) -> List[ReportProductComplianceSchema]:
+        compliance_product_list: List[ReportProductComplianceSchema] = []
+        report_products = (
+            ReportProduct.objects.order_by("product_id")
+            .filter(report_version_id=report_version_id)
+            .distinct('product_id')
+        )
+        for rp in report_products:
+            ei = ProductEmissionIntensity.objects.get(
+                product_id=rp.product_id
+            ).product_weighted_average_emission_intensity
+            industrial_process = ReportProductService.get_allocated_emissions_by_report_product_emission_category(
+                report_version_id, rp.product_id, [3]
+            )  # ID=3 is Industrial Emissions category
+            production_totals = ReportProductService.get_report_product_aggregated_totals(
+                report_version_id, rp.product_id
+            )
+            total_allocated = ReportProductService.get_allocated_emissions_by_report_product_emission_category(
+                report_version_id, rp.product_id, list(EmissionCategory.objects.all().values_list('id', flat=True))
+            )
+            allocated_reporting_only = ReportProductService.get_reporting_only_allocated(
+                report_version_id, rp.product_id
+            )
+            allocated_for_compliance = total_allocated - allocated_reporting_only
+            allocated_for_compliance_2024 = (
+                allocated_for_compliance / Decimal(production_totals["annual_amount"])
```

Comment:
I'm casting these to Decimal for now, but I think the "annual_amount" & "apr_dec_amount" in the report_production model should be Decimals not floats. Just making a note to myself to fix it later :)